src/message/messageSelectors.js

```diff
@@ -18,7 +19,30 @@ import { NULL_ARRAY } from '../nullObjects';
 export const getPrivateMessages: Selector<Message[]> = createSelector(
   getAllNarrows,
   getMessages,
-  (narrows, messages) => (narrows[ALL_PRIVATE_NARROW_STR] || NULL_ARRAY).map(id => messages[id]),
+  (narrows, messages) => {
+    const privateMessages: Message[] = [];
+    const unknownIds: number[] = [];
+
+    const pmIds = narrows[ALL_PRIVATE_NARROW_STR] || NULL_ARRAY;
+    pmIds.forEach(id => {
+      const msg = messages[id];
+      if (msg !== undefined) {
+        privateMessages.push(msg);
+      } else {
+        unknownIds.push(id);
+      }
+    });
+
+    // BUG: all messages in `narrows` _should_ also be in `messages`.
+    // Error reports indicate that, somehow, this isn't always so.
+    if (unknownIds.length > 0) {
+      logging.error('narrow IDs not found in state.messages', {
+        all_ids: pmIds,
```

Comment:
This seems like it could be a lot of data, and it's not obvious it'd be essential in studying the issue. Perhaps just the length?

Reply:
I suspect we may want a little more information than that. Trimmed to also provide the first and last 10 entries.